pyulog/core.py

```diff
@@ -648,17 +717,23 @@ def _read_file_data(self, message_name_filter_list, read_until=None):
                 data_item = ULog.Data(value)
                 self._data_list.append(data_item)
 
-    def _check_file_corruption(self, header):
-        """ check for file corruption based on an unknown message type in the header """
-        # We need to handle 2 cases:
-        # - corrupt file (we do our best to read the rest of the file)
-        # - new ULog message type got added (we just want to skip the message)
+    def _check_packet_corruption(self, header):
+        """
+        check for data corruption based on an unknown message type in the header
+        set _file_corrupt flag to true if a corrupt packet is found
+        We need to handle 2 cases:
+        - corrupt file (we do our best to read the rest of the file)
+        - new ULog message type got added (we just want to skip the message)
+        return true if packet associated with header is corrupt, else return false
+        """
+        data_corrupt = False
         if header.msg_type == 0 or header.msg_size == 0 or header.msg_size > 10000:
             if not self._file_corrupt and self._debug:
                 print('File corruption detected')
+            data_corrupt = True
             self._file_corrupt = True
 
-        return self._file_corrupt
```

Comment:
@bkueng I don't know if this was intentional earlier or a bug, but the result of this function was sticky, in that ones a corrupt packet was found this function always returned true, so the logic to 'ignore' new msg_type packets later in the file would no longer work. This did not work well with sync_recovery, so I changed this function to return true only if that particular packet was found to be corrupt.

Reply:
Indeed, that was not intended.